runtime/src/lib.rs

```diff
@@ -891,8 +891,41 @@ pub type SignedPayload = generic::SignedPayload<Call, SignedExtra>;
 /// Extrinsic type that has already been checked.
 pub type CheckedExtrinsic = generic::CheckedExtrinsic<AccountId, Call, SignedExtra>;
 
+/// Custom runtime upgrade to execute the balances migration before the account migration.
+mod custom_migration {
+    use super::*;
+
+    use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
+    use pallet_staking::migrations::clear_slash_data;
+    use hex_literal::hex;
+    use sp_runtime::{DispatchResult};
+
+    pub struct Upgrade;
+    impl OnRuntimeUpgrade for Upgrade {
+        fn on_runtime_upgrade() -> Weight {
+            // This will remove all the slash related data on staking pallet
+            clear_slash_data::<Runtime>();
```

Comment:
If at the time of the upgrade, there are some unapplied slashes they will be deleted as well, right? No objection, just informational.

Reply:
yeap this will remove all the data related to slash